cms/src/main/java/uk/nhs/digital/ps/PublicationDateValidator.java

```diff
@@ -0,0 +1,201 @@
+                "Cannot validate field ''{0}'' of type ''{1}''. Expected field of type ''{2}''.",
+                fieldValidator.getFieldDescriptor().getPath(),
+                actualFieldTypeName,
+                DATE_FIELD_TYPE_NAME
+            ));
+        }
+
+        final String actualFieldNodePathName = fieldValidator.getFieldDescriptor().getPath();
+        if (!(DATE_PROPERTY_NAME.equals(actualFieldNodePathName))) {
+            throw new ValidationException(format(
+                "Cannot find property {0} in node ''{1}'' with path ''{2}''.",
+                DATE_PROPERTY_NAME,
+                fieldValidator.getFieldDescriptor().getPath(),
+                actualFieldNodePathName
+            ));
+        }
+    }
+
+    @Override
+    public Set<Violation> validate(final IFieldValidator fieldValidator,
+                                   final JcrNodeModel documentModel,
+                                   final IModel coverageDateFieldModel
+    ) throws ValidationException {
+
+        final Set<Violation> violations = new HashSet<>();
+
+        if (isPublicationFinalised(documentModel)) {
+            return violations;
+        }
+
+        final Calendar nominalDateRaw = getDatePropertyValue(documentModel, DATE_PROPERTY_NAME);
+        final ZoneId timeZoneId = nominalDateRaw.getTimeZone().toZoneId();
+
+        // When user uses timezone which honours daylight savings, dates that fall within the period
+        // when daylight savings are in force get saved as 'previous' day, with time set
+        // to 23:00. Date falling outside this period get saved with time set to midnight of the
+        // 'current' day (i.e. with neither day or time value being affected). It's irrelevant
+        // whether the moment the user makes the update happens while daylight savings are in force.
+        //
+        // For example, with user in timezone Europe/London:
+        // user within BST:     2018-06-14 (BST date) gets saved in JCR as 2018-06-13T23:00:00Z
+        // user within BST:     2018-01-14            gets saved in JCR as 2018-01-14T00:00:00Z
+        // user outside of BST: 2018-06-14 (BST date) gets saved in JCR as 2018-06-13T23:00:00Z
+        // user outside of BST: 2018-01-14            gets saved in JCR as 2018-01-14T00:00:00Z
+        //
+        // In order to be able to easily compare the dates on day component (without taking time
+        // component into account as actual time component doesn't get persisted anyway),
+        // we need to compensate for potential daylight savings offset. Adding an hour to the
+        // value read from JCR before truncating the time component achieves that, bringing the
+        // 'decremented'/daylight saving dates into line while not affecting the non-daylight
+        // saving ones (actually, it does but the extra hour will get truncated anyway so it doesn't
+        // matter).
+        final LocalDate nominalDate = nominalDateRaw.toInstant()
```

Comment:
Can you not just get the date after the time zone conversion which should then be correct?